src/main/proto/com/google/api/codegen/config.proto

```diff
@@ -122,6 +122,9 @@ message LanguageSettingsProto {
 
   // The release level of the client in the language
   ReleaseLevel release_level = 6;
+
+  // Language specific flags to control fine-grained properties in package managers
+  map<string, string> packaging_settings = 7;
```

Comment:
I mean to say that this field could changed to `string package_name = 7` which would provide more documentation of the configuration.

Reply:
The issue with that is that there is already an `string package_name = 1` but that's not the same thing as the gem name.

Reply:
Gah, unfortunately thats a bit of a misnomer. We could call it `metadata_identifier`.

Reply:
I considered that too. The issue here is now we're projecting gem_name (or whatever we call it) on to all supported languages and I'm not sure that makes sense. i.e. how do I know that a metadata_identifier is the gem name for Ruby? What is it for C#, etc.?

To me, the extra indirection is at least as confusing as creating an arbitrary map of strings.

Reply:
> how do I know that a metadata_identifier is the gem name for Ruby

I think the documentation can be both general enough to cover multiple languages but precise enough for one to imply that this field will override the gem_name that is inferred from the package_name field.

 Furthermore we can guard all of the languages that this doesn't make sense for by throwing an error by default if that field is set.

Reply:
@lukesneeringer thoughts on this?